### DISCUSSION TITLE
Allow fall through in method decleration

### DISCUSSION THREAD

**Discussion by lazyloader** [dotnet/csharplang#1664] [Language Ideas] [answered]
Instead of this code

```csharp
        public override void Method1()  => SomeMethod();
        public override void Method2() => SomeMethod();
        public override void Method3()  => SomeMethod();
```

We can do this code:
```csharp
        public override void Method1()  =>
        public override void Method2() =>
        public override void Method3()  => SomeMethod();
```

and it will automatically go through so they all execute `SomeMethod`

this would be useful when implementing the state pattern where one state would throw the same error "invalid operation" in the cases where something was attempted to be used but in the incorrect state.

**Comment by HaloFour** [comment, marked as answer]
It would only be useful if every method took the same parameters and only differed semantically based on name yet still performed the exact same logic.  In other words, not useful at all.

**Comment by jnm2** [comment]
'decleration' is a typo.

**Comment by MgSam** [comment]
This guy is having a blast trolling the C# forums.

**Comment by DavidArno** [comment]
> ... this would be useful when implementing the state pattern ...

If you need to request a language change because of a pattern, then don't use that pattern in the first place. Implement that state model differently.

**Comment by Unknown6656** [comment]
I'll reference my [lengthy comment here](https://github.com/dotnet/csharplang/issues/1662#issuecomment-399776344) (from issue #1662), as I have not the patience to re-type nearly the same thing:

### Why?
#### Why would we (or you) need this?
If you have a function with the same semantics under multiple names, then why not use the first function on every occurrence?

-------

@MgSam : I wish it was so, but I (still) believe that the guy is totally serious about this....

But damn, I did not have so much fun for quite some time, I have to admit.....
My favourite request, however, was this one #993 (translation: https://translate.google.com/translate?tl=en&u=https://github.com/dotnet/csharplang/issues/993), which requested a mandarin version of C#.
